pkg/cmd/cli/config/smart_merge.go

```diff
@@ -97,7 +97,11 @@ func CreateConfig(namespace string, clientCfg *client.Config) (*clientcmdapi.Con
 	config.AuthInfos[userNick] = *credentials
 
 	cluster := clientcmdapi.NewCluster()
-	cluster.Server = clientCfg.Host
+	if normalizedServerURL, err := NormalizeServerURL(clientCfg.Host); err == nil {
```

Comment:
`clientCfg.Host` should already be known to work at this point no matter who is calling it.  I think this is unnecessary.